client/finality-grandpa/src/communication/mod.rs

```diff
@@ -401,12 +394,14 @@ impl<B: BlockT, N: Network<B>> NetworkBridge<B, N> {
 
 		let outgoing = outgoing.with(|out| {
 			let voter::CommunicationOut::Commit(round, commit) = out;
-			Ok((round, commit))
+			future::ok((round, commit))
 		});
 
 		(incoming, outgoing)
 	}
+}
 
+impl<B: BlockT, N: Network<B>> NetworkBridge<B, N> {
```

Comment:
Doesn't this have the same bounds as the impl above? Why create a new one just for this function.